### PR TITLE
devdocs: documentation of logging conventions

### PR DESCRIPTION
**Pull request by daniel-weisse**
### Proposed change(s)

Adds documentation for logging conventions in our code.
Not sure if I wrote too much for quick documentation or if there is still information missing that should be there.
Also, this mostly covers my personal opinions on logging, so there might be some disagreements I would be happy to discuss.

### CODE REVIEW

File: .github/docs/conventions.md [hunk 1]
```diff
@@ -55,6 +55,61 @@ golangci-lint run ./...
...
+  log.Infof("Waiting to do something...")
+  log.Error("A critical error occurred!")
+  ```
+
+* Use the `With()` method to add structured context to your log messages. The context tags should be easily searchable to allow for easy log filtering.
+  
+  Example:
+
+  ```Go
+  log.With(zap.Error(someError), zap.String("endpoint", "192.0.2.1")).Errorf("Connecting to remote endpoint failed")
```
Comment (katexochen) [inline .github/docs/conventions.md:84]:
```suggestion
  log.With(zap.Error(someError), zap.String("ip", "192.0.2.1")).Errorf("Connecting to IP failed")
```

**Review by katexochen — APPROVED** [review]
Hope we can soon replace this with official Go stuff...

**Comment by daniel-weisse** [comment]
>Hope we can soon replace this with official Go stuff...

We already have usage instructions in the logging package documentation (`/internal/logger/log.go`).
If the other developers agree that this is the correct, or better place, to keep logging conventions, I am happy to move everything from this PR to there.

**Comment by katexochen** [comment]
> > Hope we can soon replace this with official Go stuff...

> We already have usage instructions in the logging package documentation...

Sorry, my comment was a bit short. What I've meant was that I'm looking forward to replace the whole logging package including documentation with the new [`slog`](https://cs.opensource.google/go/x/exp/+/master:slog/doc.go) package, which will likely provide structured logging to the standard library in the future and is currently in experimental state.

**Review by malt3 — APPROVED** [review]

File: .github/docs/conventions.md [hunk 2]
```diff
@@ -55,6 +55,61 @@ golangci-lint run ./...
...
+We use a [custom subset](/internal/logger/) of [zap](https://pkg.go.dev/go.uber.org/zap) to provide logging for Constellation's services and components.
+Usage instructions can be found in the package documentation.
+
+Certain components may further specify a subset of the logger for their use. For example, the CLI has a debug-only logger, restricting the use of the logger to only `Debugf()`.
+
+Further we try to adhere to the following guidelines:
+
+* Do not log potentially sensitive information, e.g. variables that contain keys, secrets or otherwise protected information.
+
+* Start log messages with in uppercase and end without a punctuation mark. Exclamation, question marks, or ellipsis may be used where appropriate.
```
Comment (derpsteb) [inline .github/docs/conventions.md:69]:
```suggestion
* Start log messages in uppercase and end without a punctuation mark. Exclamation, question marks, or ellipsis may be used where appropriate.
```

**Review by derpsteb — APPROVED** [review]

**Pull request merged**